### PR TITLE
Convert question render library to async

### PR DESCRIPTION
**Pull request by jonatanschroeder**
Follow-up to #9329. Converts remaining code in question-render.js to async. Also:

* Converts renderPanelsForSubmission function so it receives named parameters instead of 9 ordered ones;
* Provides better types for arguments of related functions;
* A few more query function upgrades with schema;
* Removed unused columns from queries;
* Removed unused parameter from question servers render functions.

### CODE REVIEW

File: apps/prairielearn/src/lib/question-render.sql [hunk 1]
```diff
@@ -72,11 +72,7 @@ SELECT
   format_date_full_compact (
     s.date,
     coalesce(ci.display_timezone, c.display_timezone)
-  ) AS formatted_date,
-  CASE
-    WHEN s.grading_requested_at IS NOT NULL THEN format_interval ($req_date - s.grading_requested_at)
-    ELSE NULL
-  END AS elapsed_grading_time
```
Comment (jonatanschroeder) [inline apps/prairielearn/src/lib/question-render.sql:79]: This column was never used anywhere.

File: apps/prairielearn/src/lib/question-render.sql [hunk 2]
```diff
@@ -188,7 +184,6 @@ SELECT
   to_jsonb(ci) AS course_instance,
   to_jsonb(c) AS variant_course,
   to_jsonb(qc) AS question_course,
-  to_jsonb(ci) AS course_instance,
```
Comment (jonatanschroeder) [inline apps/prairielearn/src/lib/question-render.sql:191]: This is duplicate, it's the same as three lines up.

File: apps/prairielearn/src/lib/db-types.ts [hunk 3]
```diff
@@ -849,3 +849,25 @@ export const IssueSchema = z.object({
...
+// Result of grading_job_status sproc
+export const GradingJobStatusSchema = z.enum([
+  'none',
+  'canceled',
+  'queued',
+  'grading',
+  'graded',
+  'requested',
+]);
+export type GradingJobStatus = z.infer<typeof GradingJobStatusSchema>;
```
[the remark this answers is not in the document]
Comment (nwalters512) [reply apps/prairielearn/src/lib/db-types.ts:873]: I think this is fine for now.

File: apps/prairielearn/src/pages/instructorAssessmentManualGrading/instanceQuestion/instanceQuestion.ts [hunk 4]
```diff
@@ -77,17 +76,17 @@ router.get(
...
-      res.locals.question.id,
-      res.locals.instance_question.id,
-      req.params.variant_id,
-      res.locals.urlPrefix,
-      null, // questionContext
-      null, // csrfToken
-      null, // authorizedEdit
-      false, // renderScorePanels
-    );
+    const results = await renderPanelsForSubmission({
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorAssessmentManualGrading/instanceQuestion/instanceQuestion.ts:79]: This kind of API is so much nicer than a function with 9 separate parameters!

**Review by nwalters512 — APPROVED** [review]

**Pull request merged**